src/System.Private.CoreLib/shared/System/Buffers/Text/Utf8Parser/Utf8Parser.Float.cs

```diff
@@ -113,31 +115,65 @@ private static bool TryParseNormalAsFloatingPoint(ReadOnlySpan<byte> source, ref
         //
         private static bool TryParseAsSpecialFloatingPoint<T>(ReadOnlySpan<byte> source, T positiveInfinity, T negativeInfinity, T nan, out T value, out int bytesConsumed)
         {
-            if (source.Length >= 8 &&
-                source[0] == 'I' && source[1] == 'n' && source[2] == 'f' && source[3] == 'i' &&
-                source[4] == 'n' && source[5] == 'i' && source[6] == 't' && source[7] == 'y')
-            {
-                value = positiveInfinity;
-                bytesConsumed = 8;
-                return true;
-            }
+            int srcIndex = 0;
+            bool isNegative = false;
+            byte c = source[srcIndex];
```

Comment:
Is source guaranteed to be non-empty here?

Reply:
No, we don't have a check for that in the caller; will update

Reply:
Fixed.